BoxContentSDK/BoxContentSDK/Requests/BOXItemSetCollectionsRequest.m

```diff
@@ -50,53 +78,122 @@ - (instancetype)initFolderSetCollectionsRequestForFolderWithID:(NSString *)folde
+                                                     collectionIDs:(NSArray *)collectionIDs
+{
+    return [self initWithItemID:bookmarkID
+                  collectionIDs:collectionIDs
+                       resource:BOXAPIResourceBookmarks];
+}
+
 - (instancetype)initBookmarkSetCollectionsRequestForBookmarkWithID:(NSString *)bookmarkID
                                                      collectionIDs:(NSArray *)collectionIDs
+                                                       associateId:(NSString *)associateId
 {
-        return [self initWithItemID:bookmarkID
-                      collectionIDs:collectionIDs
-                           resource:BOXAPIResourceBookmarks];
+    return [self initWithItemID:bookmarkID
+                       resource:BOXAPIResourceBookmarks
+                  collectionIDs:collectionIDs
+                    associateId:associateId];
 }
 
-- (BOXAPIOperation *)createOperation
+- (BOXAPIAuthenticatedOperation *)createOperation
 {
-    BOXAPIJSONOperation *operation = nil;
-    
-    NSURL *url = [self URLWithResource:self.resource 
+    NSURL *url = [self URLWithResource:self.resource
                                     ID:self.itemID
                            subresource:nil
                                  subID:nil];
-
+    
     NSDictionary *queryParameters = nil;
     if (self.requestAllItemFields) {
         queryParameters = @{BOXAPIParameterKeyFields :[self fullItemFieldsParameterString]};
     }
-
+    
     NSMutableArray *bodyContent = [NSMutableArray array];
+    
     for (NSString *collectionID in self.collectionIDs) {
         [bodyContent addObject:@{BOXAPIObjectKeyID : collectionID}];
     }
     NSDictionary *bodyDictionary = @{BOXAPIObjectKeyCollections : bodyContent};
     
-    operation = [self JSONOperationWithURL:url 
-                                HTTPMethod:BOXAPIHTTPMethodPUT
-                     queryStringParameters:queryParameters
-                            bodyDictionary:bodyDictionary
-                          JSONSuccessBlock:nil
-                              failureBlock:nil];
-    
-    return operation;
+    if (self.associateId != nil) {
+        
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:url
+                                                             HTTPMethod:BOXAPIHTTPMethodPUT
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:bodyDictionary
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateId];
+        
+        if (self.associateId != nil) {
+             NSString *destinationPath = [NSTemporaryDirectory() stringByAppendingPathComponent:self.associateId];
```

Comment:
Your right, initially I had the path forwarded from our services and then encapsulated the file as a container for background tasks. But point taken, simply migrating the temp directory to the shared container will prevent 3rd party developers needed flexibility.